service.go

```diff
@@ -376,6 +376,14 @@ func (client *Client) GetService(id ID) (*Service, error) {
 	return &q.Account.Service, nil
 }
 
+func (client *Client) GetService(identifier string) (*Service, error) {
```

Comment:
we've been meaning to do this for a LONG time and with the 2025 release emmient - now is a good time to pull the trigger on this breaking change.

For opslevel-go using customers they won't automatically upgrade to this version so it won't break them because the v2025 release is like a "major" version change.  So they'll have to explicitly upgrade and even then the go compiler should catch most of these changes